Sources/MongoSwift/BSON/AnyBSONValue.swift

```diff
@@ -103,4 +103,165 @@ public struct AnyBSONValue: Codable, Equatable {
                 message: "Encountered a value that could not be decoded to any BSON type")
         }
     }
+
+    public func hash(into hasher: inout Hasher) {
+        switch self.value {
+        case let value as Int:
+            value.hash(into: &hasher)
+        case let value as Int32:
+            value.hash(into: &hasher)
+        case let value as Int64:
+            value.hash(into: &hasher)
+        case let value as Double:
+            value.hash(into: &hasher)
+        case let value as Decimal128:
+            value.hash(into: &hasher)
+        case let value as Bool:
+            value.hash(into: &hasher)
+        case let value as String:
+            value.hash(into: &hasher)
+        case let value as RegularExpression:
+            value.hash(into: &hasher)
+        case let value as Timestamp:
+            value.hash(into: &hasher)
+        case let value as Date:
+            value.hash(into: &hasher)
+        case let value as MinKey:
+            value.hash(into: &hasher)
+        case let value as MaxKey:
+            value.hash(into: &hasher)
+        case let value as ObjectId:
+            value.hash(into: &hasher)
+        case let value as CodeWithScope:
+            value.hash(into: &hasher)
+        case let value as Document:
+            value.hash(into: &hasher)
+        case let value as Binary:
+            value.hash(into: &hasher)
+        case let value as [BSONValue?]:
+            value.forEach({$0.map({AnyBSONValue($0).hash(into: &hasher)})})
+        default: break
+        }
+    }
+
+    /// This value as a BsonDocument if it is one, otherwise nil
+    public lazy var asDocument = value as? Document
```

Comment:
idk if I'm sold on adding all of this to the public API - is `x.asDocument` that much of an improvement over `x.value as? Document`? 
if we keep them I would like to see all the docstrings formatted like this:
```swift
/// The underlying value as a `Document` if it is one, otherwise `nil`.
```

Reply:
I was going by a conversation that was had in #swift– I'm happy to remove it.

Reply:
It doesn't have much value add. IMO, we should either make this into an enum, or remove these.